salt/key.py

```diff
@@ -373,7 +373,7 @@ class Key(object):
     REJ = 'minions_rejected'
     DEN = 'minions_denied'
 
-    def __init__(self, opts, io_loop=None):
+    def __init__(self, opts):
```

Comment:
The reason we had this was to correct https://github.com/saltstack/salt/issues/38209

Are we certain that we won't regress here?

Reply:
I'll check this case when I'll done here.
Anyway Tornado doesn't allow to play with IOLoops now and we shall use the current one everywhere.

Reply:
> Anyway Tornado doesn't allow to play with IOLoops now and we shall use the current one everywhere.

But we should have the ability to use older ones.

Reply:
I understand this, but If we use the only on IOLoop per thread/process, we should be OK with Tornado 4.
Anyway I'll take an additional attention on this.

Reply:
@DmitryKuzmenko yeah, I think so too. But just keep in mind that. Sometimes you can have pretty old systems in the enterprise, but who want to have latest possible codebase (with outdated dependencies, of course)... :wink: